lib/sof/cycle/parser.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/hash/keys"
```

Comment:
:thought_balloon: This is a heavy set of requirements on Rails/ActiveRecord. Would it be possible to decouple this gem from Rails so it's pure Ruby since Rails is so heavyweight for a gem? For example, you the [Refinements](https://alchemists.io/projects/refinements) gem is perfect for situations like this.

Reply:
This was extracted from a rails app. Yes, it would be good to limit dependencies but it's not a high priority at the moment